dev/how-to/get-started/data-migration.md

```diff
@@ -0,0 +1,722 @@
+1729382256910270828     bac9162b47c56fc8a4d2a519803d51b3        3309
+1729382256910270829     9be40cee5b0eee1462c82c6964087ff9        3309
+1729382256910270830     5ef698cd9fe650923ea331c15af3b160        3309
+1729382256910270831     05049e90fa4f5039a8cadc6acbb4b2cc        3309
+1729382256910270832     cf004fdc76fa1a4f25f62e0eb5261ca3        3309
+1729382256910270833     0c74b7f78409a4022a2c4c5a5ca3ee19        3309
+1729382256910270834     d709f38ef758b5066ef31b18039b8ce5        3309
+1729382256910270835     41f1f19176d383480afa65d325c06ed0        3309
+1729382256910270836     24b16fede9a67c9251d3e7c7161c83ac        3309
+```
+
+DM 使用一种算法将上游 MySQL 实例分配的 ID 进行位移，以生成下游 TiDB 实例的唯一 ID。在测试中，partition ID 仅由 "instance ID" 组成，因为上游 MySQL Server 中 schema 和表名是相同的。我们将 partition ID 表达式参数中的 "schema ID" 和 "table ID" 留空：
+
+```bash
+grep arguments dm-cnf/dmtask2.yaml
+```
+
+输出结果：
+
+```
+    arguments: ["1", "", ""]
+    arguments: ["2", "", ""]
+    arguments: ["3", "", ""]
+```
+
+上游 MySQL Server 分配的最后一个自增 ID 是 372。数据迁移到 TiDB server 后，转换后的自增 ID 最大值来自 instance 3（对应从右向左第一列的 3309 端口）。中间一列最后一行的值与 MySQL 实例中自增 ID 372 所在行的值相同。这个算法为来自上游的自增 ID 分配 64 位整数的 44 位，这意味着 partition ID 列映射方案的默认实现无法处理大于 2^44（大约 17.5 万亿）的值。1 位为符号的保留位，4 位为 instance ID，7 位为 schema ID，8 位为 table ID（44 + 1 + 4 + 7 + 8 = 64 位）。
+
+在此，可为本示例复现算法，以自增 ID 372、instance ID 为 3 为例：
+
+```bash
+id=372 instance_id=3 schema_id=0 table_id=0
+echo $(( instance_id << (64-1-4) | schema_id << (64-1-4-7) | table_id << 44 | id ))
+```
+
+输出结果：
+
+```
+1729382256910270836
+```
+
+因为只有 44 位对应原始的自增值，所以可以舍弃其他位，从而将转换后的值转换回原来的值：
+
+```bash
+echo $(( 1729382256910270836 & (1<<45)-1 ))
+```
+
+输出结果：
+
+```
+372
+```
+
+可使用 SQL 查询中的表达式查看转换后的 ID 和原来的 ID：
+
+```bash
+mysql -h 127.0.0.1 -P 4000 -u root -e 'select id, id&(1<<45)-1 as orig_id, c, port from t1 order by orig_id' dmtest2 | tail
+```
+
+输出结果：
+
+```
+576460752303423857      369     0c74b7f78409a4022a2c4c5a5ca3ee19        3307
+576460752303423858      370     d709f38ef758b5066ef31b18039b8ce5        3307
+1152921504606847346     370     d709f38ef758b5066ef31b18039b8ce5        3308
+1729382256910270834     370     d709f38ef758b5066ef31b18039b8ce5        3309
+576460752303423859      371     41f1f19176d383480afa65d325c06ed0        3307
+1729382256910270835     371     41f1f19176d383480afa65d325c06ed0        3309
+1152921504606847347     371     41f1f19176d383480afa65d325c06ed0        3308
+576460752303423860      372     24b16fede9a67c9251d3e7c7161c83ac        3307
+1729382256910270836     372     24b16fede9a67c9251d3e7c7161c83ac        3309
+1152921504606847348     372     24b16fede9a67c9251d3e7c7161c83ac        3308
+```
+
+## 结论
+
+本教程完成了 2 个练习：
+
+- 上游 3 个 MySQL Server 实例的分片迁移，每个实例的自增 ID 都互不重叠。
+- 上游 3 个 MySQL Server 实例的分片迁移，这些实例的自增 ID 存在冲突。
+
+本教程介绍了分片迁移中，DM 如何在集群中导入初始数据，以及 DM 读取 binlog，从而使下游 TiDB 集群与上游实例保持同步。
+
+关于 DM 的更多详情，请参考 [Data Migration 简介](/reference/tools/data-migration/overview.md)，或加入 [TiDB Community Slack](https://pingcap.com/tidbslack/) channel 参与讨论。
```

Comment:
The link is broken.

Reply:
Thanks. But I find these links available, and this PR has passed the link check CI.